tensorcircuit/templates/ensemble.py

```diff
@@ -0,0 +1,165 @@
+"""
+Useful utilities for ensemble
+"""
+
+import tensorflow as tf
+import numpy as np
+
+
+class bagging:  # A.K.A. voting
+    def __init__(self):
+        self.models = []
+        self.is_trained = []
+        self.count = 0
+        self.need_confidence = True  # Help in reducing numbers of get_confidence runs
+        self.permit_train = False
+
+    def append(self, model, is_trained):
+        """
+        Add model to the voting method
+        """
+        self.models.append(model)
+        self.is_trained.append(is_trained)
+        self.count += 1
+
+    def __train_model(self, i, **kwargs):
+        """
+        Train a model if it isn't trained already
+        """
+        if not self.is_trained[i]:
+            self.need_confidence = True
+            self.is_trained[i] = True
+            self.models[i]
+            self.models[i].trainable
```

Comment:
these two lines seems to be redundant?

Reply:
You have highlighted four lines, and which two lines in specific that you are suggesting a possible redundancy?

Reply:
> You have highlighted four lines, and which two lines in specific that you are suggesting a possible redundancy?

L32 and L33

Reply:
L39 now is still redundant?